codebreaker_garage/lib/codebreaker_garage/game.rb

```diff
@@ -0,0 +1,75 @@
+module Codebreaker_garage
+
+  class Game
+
+    ATTEMPTS = 9
+    HINTS = 3
+
+    attr_reader :attempts, :hints
+
+    def initialize
+      @generated_code = generate_code
+      @attempts = ATTEMPTS
+      @hints = HINTS
+    end
+
+    def has_attempts?
+      @attempts != 0
+    end
+
+    def statistics
+      {
+        time: Time.now,
+        secret_code: @generated_code.to_s,
+        attempts: ATTEMPTS,
+        hints: HINTS,
+        user_attempts_left: @attempts,
+        user_hints_left: @hints
+      }
+    end
+
+    def receive_hint
+      return "You have no hints." if @hints == 0
+      @hints -= 1
+      machine_codebreaker = (1..6).to_a
+      hint = machine_codebreaker.repeated_permutation(4).to_a
+      hint.select! {|code| code == @generated_code }
```

Comment:
`machine_codebreaker = (1..6).to_a `
`hint = machine_codebreaker.repeated_permutation(4).to_a`
`hint.select! {|code| code == @generated_code }`
This part of code has same result as ` @generated_code[rand(4)]`
I just was experementing with method `repeated_permutation` which generates all possible variants of code.